dazel.py

```diff
@@ -43,41 +44,46 @@ class DockerInstance:
     set it up through configuration variables, and pass on commands to it.
     It streams the output directly and blocks until the command finishes.
     """
-    
+
     def __init__(self, instance_name, image_name, run_command, dockerfile,
                        repository, directory, command, volumes, ports, network,
                        run_deps, docker_compose_file, docker_compose_project_name,
                        docker_compose_services, bazel_user_output_root, bazel_rc_file,
-                       docker_run_privileged, docker_machine, dazel_run_file):
+                       docker_run_privileged, docker_machine, dazel_run_file, workspace_hex):
         real_directory = os.path.realpath(directory)
-        self.workspace_hex_digest = hashlib.md5(real_directory.encode("ascii")).hexdigest()
-
-        self.instance_name = "%s_%s" % (instance_name, self.workspace_hex_digest)
-        self.image_name = "%s_%s" % (image_name, self.workspace_hex_digest)
+        self.workspace_hex_digest = ""
+        self.instance_name = instance_name
+        self.image_name = image_name
         self.run_command = run_command
         self.dockerfile = dockerfile
         self.repository = repository
         self.directory = directory
         self.command = command
         self.network = network
         self.docker_compose_file = docker_compose_file
-        self.docker_compose_project_name = "%s%s" % (docker_compose_project_name,
-                                                     self.workspace_hex_digest)
+        self.docker_compose_project_name = docker_compose_project_name
         self.bazel_user_output_root = bazel_user_output_root
         self.bazel_output_base = ""
         self.bazel_rc_file = bazel_rc_file
         self.docker_run_privileged = docker_run_privileged
         self.docker_machine = docker_machine
         self.dazel_run_file = dazel_run_file
 
+        if workspace_hex:
+            self.workspace_hex_digest = hashlib.md5(real_directory.encode("ascii")).hexdigest()
+            self.instance_name = "%s_%s" % (self.instance_name, self.workspace_hex_digest)
+            self.image_name = "%s_%s" % (self.image_name, self.workspace_hex_digest)
```

Comment:
This was actually an error I made - the image should only be different if there is a dockerfile to build from.

Someone already complained about this when you are just pulling dazel/dazel since there is no such image with hex.